storage/src/main/java/io/delta/storage/S3SingleDriverLogStore.java

```diff
@@ -264,10 +264,13 @@ public void write(
                 final CountingOutputStream stream =
                     new CountingOutputStream(fs.create(resolvedPath, overwrite));
 
-                while (actions.hasNext()) {
-                    stream.write((actions.next() + "\n").getBytes(StandardCharsets.UTF_8));
+                try {
+                    while (actions.hasNext()) {
+                        stream.write((actions.next() + "\n").getBytes(StandardCharsets.UTF_8));
+                    }
+                } finally {
+                    stream.close();
```

Comment:
I don't _think_ we want to do this.

`CountingOutputStream::close` will flush contents to output and then close the output stream. If an exception happened, we don't want the contents to be flushed. That may create a partial file.

Reply:
@tcw Closing the PR based on the above comment. Feel free to make a PR to add a comment about why we are not closing the stream on error.